### ISSUE TITLE
[REVIEW]: IndividualDisplacements.jl: a Julia package to simulate and study particle displacements within the climate system

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2813]
**Submitting author:** @gaelforget (<a href="http://orcid.org/0000-0002-4234-056X">Gael Forget</a>)
**Repository:** <a href="https://github.com/JuliaClimate/IndividualDisplacements.jl" target ="_blank">https://github.com/JuliaClimate/IndividualDisplacements.jl</a>
**Version:** v0.2.0
**Editor:** @kthyng
**Reviewer:** @milankl, @visr
**Archive:** <a href="https://doi.org/10.5281/zenodo.4678742" target="_blank">10.5281/zenodo.4678742</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/f61c861699db08f38b2512818a4862db/status.svg)](https://joss.theoj.org/papers/f61c861699db08f38b2512818a4862db)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/f61c861699db08f38b2512818a4862db"><img src="https://joss.theoj.org/papers/f61c861699db08f38b2512818a4862db/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/f61c861699db08f38b2512818a4862db/status.svg)](https://joss.theoj.org/papers/f61c861699db08f38b2512818a4862db)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/JuliaClimate/IndividualDisplacements.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@milankl & @visr, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @kthyng know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaClimate/IndividualDisplacements.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@gaelforget) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaClimate/IndividualDisplacements.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@gaelforget) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @milankl, @visr it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2813 with the following error: 

 Can't find any papers to compile :-(

**Comment by kthyng** [comment]
@milankl, @visr this is where the actual review takes place. As you go through the checklist and the [JOSS reviewer docs](https://joss.readthedocs.io/en/latest/reviewer_guidelines.html), open issues in the software repository, then link to this review issue for bookkeeping. That way, this issue stays relatively clean but has clear links out to all related review issues and they can be tracked independently in other issues. Let me know if you have any questions and otherwise I'll check in periodically.

**Comment by kthyng** [comment]
@whedon generate pdf from branch josspaper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch josspaper. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2813 with the following error: 

 Can't find any papers to compile :-(

**Comment by kthyng** [comment]
@whedon generate pdf from branch josspaper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch josspaper. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2813 with the following error: 

 Can't find any papers to compile :-(

**Comment by kthyng** [comment]
@whedon generate pdf from branch josspaper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch josspaper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:wave: @milankl, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @visr, please update us on how your review is going.

**Comment by milankl** [comment]
Hey Kristen @kthyng, the invitation expired. Could you send a new one?

**Comment by kthyng** [comment]
@whedon re-invite @milankl as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@milankl please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by milankl** [comment]
Works fine. Thanks!

**Comment by kthyng** [comment]
Hi @milankl, @visr! Just a friendly ping to see how your reviews are coming along. Do you have an estimate for when you can work on this again?

Also a ping to @gaelforget since I see you are working on some review issues that have come in.

**Comment by visr** [comment]
I plan to work on this again next week. Quite a few improvements have come along since I had a look the first time.

**Comment by visr** [comment]
I'm basically done with review, only I don't yet feel confident to tick off the "Functionality" and "Functionality Documentation" boxes just yet, but that is being discussed in https://github.com/JuliaClimate/IndividualDisplacements.jl/issues/38.

**Comment by kthyng** [comment]
Thanks @visr. 

@milankl How are things coming along in your review? Thanks!

**Comment by milankl** [comment]
Hey @kthyng, I'm basically in the same situation as Martijn, if issue [JuliaClimate/IndividualDisplacements.jl#38](https://github.com/JuliaClimate/IndividualDisplacements.jl/issues/38) gets resolved then I'm happy to tick all the boxes.

**Comment by kthyng** [comment]
@milankl ah ok, got it.

@gaelforget What is your time frame for working out the remaining review issues? I can pause this issue if you'd like more than a couple of weeks (assuming the reviewers agree).

**Comment by gaelforget** [comment]
> @milankl ah ok, got it.
> 
> @gaelforget What is your time frame for working out the remaining review issues? I can pause this issue if you'd like more than a couple of weeks (assuming the reviewers agree).

Hi @kthyng 

Am working on it now and hope to be done early next week. 

As you saw, I had closed the other, also very helpful, issues raised by @milankl and @visr in Dec. Sorry for the long break afterwards. I resumed with a bit of cleanup on the code and docs (https://github.com/JuliaClimate/IndividualDisplacements.jl/pull/52) which should help close https://github.com/JuliaClimate/IndividualDisplacements.jl/issues/38 and once done I would release v0.2.8 , merge  into there paper branch, add a summary of functionalities, and post responses on the issue tracker if that works.

Thank you all for your patience and contributions to this review so far

**Comment by kthyng** [comment]
@gaelforget Sounds perfect! Thanks for the update.

**Comment by gaelforget** [comment]
@kthyng 

Another update to let you know that I feel I have now completed the code and documentation updates (as of v0.2.9) needed to address https://github.com/JuliaClimate/IndividualDisplacements.jl/issues/38 where I just posted more detailed response to @milankl and @visr

**Comment by kthyng** [comment]
@gaelforget Ok thanks for the update. I'll wait to hear from @milankl and @visr about their status after this work.

**Comment by milankl** [comment]
Thanks @gaelforget ! Looking good at first glimpse, I'll finish the review in the next days!

**Comment by gaelforget** [comment]
> @gaelforget Ok thanks for the update. I'll wait to hear from @milankl and @visr about their status after this work.

I just posted an updated version of the paper itself (in the `josspaper_rev01` branch) to reflect code changes that have happened since original submission, and hopefully clarify the functionality summary. 

I also expended a bit on the context vs other particle tracking software as advised during the pre-review at the top of section 2 (please note that the included reference from Van Sebille et al has a lot more on this) and updated references & funding aknowledgments.

Having verified that the updated paper.md compiles and renders properly, I feel that `josspaper_rev01` is now ready to merge into the original `josspaper` branch. However, I stopped short of merging the PR as I wanted to check in with you first:

Is it a good time to merge this PR https://github.com/JuliaClimate/IndividualDisplacements.jl/pull/61 and update the `josspaper` branch or should I wait on that?

**Comment by kthyng** [comment]
@gaelforget it doesn't matter to me. If your goal is to wait to merge until the paper has absolutely no more changes, then you should wait until I do a final read through at the end of the review. Otherwise it doesn't really matter from the JOSS perspective.

**Comment by kthyng** [comment]
@milankl and @visr Will you be able to work on your reviews soon?

**Comment by gaelforget** [comment]
> @gaelforget it doesn't matter to me. If your goal is to wait to merge until the paper has absolutely no more changes, then you should wait until I do a final read through at the end of the review. Otherwise it doesn't really matter from the JOSS perspective.

Thanks for the clarification. My question came from being unsure whether I should expect detailed comments on the paper itself from reviewers and at which stage of the review process.

Accordingly, I just merged the `josspaper_rev01` into `josspaper` so everything on my end is in sync right now with the latest release (v0.2.11).

A couple follow up questions, @kthyng :

- could / should the pdf available to reviewers linked in https://github.com/openjournals/joss-reviews/issues/2813#issuecomment-723257060 be updated?
- I though about adding maybe a pair of plots illustrating results from the package (e.g. one ocean, one atmosphere); can I do that and merge in the josspaper paper branch either now or later ?

**Comment by kthyng** [comment]
> Thanks for the clarification. My question came from being unsure whether I should expect detailed comments on the paper itself from reviewers and at which stage of the review process.

Oh sorry I misunderstood your question. The reviewers should go through your paper as part of the review process as they go through the checklist so you do want it clearly available. 

> * could / should the pdf available to reviewers linked in [#2813 (comment)](https://github.com/openjournals/joss-reviews/issues/2813#issuecomment-723257060) be updated?

Yes, I can run this command now and you can run it whenever you want, too.

> * I though about adding maybe a pair of plots illustrating results from the package (e.g. one ocean, one atmosphere); can I do that and merge in the josspaper paper branch either now or later ?

I'd add it now so the reviewers can consider the paper how you intend it to be.

**Comment by kthyng** [comment]
@whedon generate pdf from branch josspaper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch josspaper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: :point_left:

**Comment by visr** [comment]
Thanks for all the updates. I just concluded the review, and checked all the boxes above.

**Comment by gaelforget** [comment]
> Thanks for all the updates. I just concluded the review, and checked all the boxes above.

Hi @visr 

This is great to hear! Thank you very much for the time you spent on this review, your patience, and the feedback you provided.

**Comment by gaelforget** [comment]
@whedon generate pdf from branch josspaper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch josspaper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: :point_left:

**Comment by gaelforget** [comment]
> 👉📄 [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02813/joss.02813/10.21105.joss.02813.pdf) 📄 [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02813/joss.02813/10.21105.joss.02813.pdf) 📄 👈

I just added a couple illustrations of the package applied to Atmosphere (#1) and Ocean (#2), plus a reference to them at the end of paragraph 1

**Comment by gaelforget** [comment]
> Thanks @gaelforget ! Looking good at first glimpse, I'll finish the review in the next days!

Hi @milankl 

Please dont hesitate to let me know if there is something else that I would need to address in order to complete the review. I am standing by, with the hope that we are ready to wrap things up

**Comment by milankl** [comment]
Hey @gaelforget, nice work! Yes, so I managed to feed IndividualDisplacements.jl with data from ShallowWaters.jl and can verify the particle advection here with the semi-Lagrangian advection scheme therein. So that ticks all the boxes from above. I'm happy for this work to get published so that hopefully more people will start using it, improving its features. There's still some issues that I may raise in the repo, but they are beyond the scope of this review. Thanks @kthyng for your efforts too.

[![000200](https://user-images.githubusercontent.com/25530332/111016072-c43a6000-83a3-11eb-9da1-794a0468fd09.png)
Click to see the video on twitter](https://twitter.com/milankloewer/status/1370561010856251393?s=20)

**Comment by gaelforget** [comment]
> Hey @gaelforget, nice work! Yes, so I managed to feed IndividualDisplacements.jl with data from ShallowWaters.jl and can verify the particle advection here with the semi-Lagrangian advection scheme therein. So that ticks all the boxes from above. I'm happy for this work to get published so that hopefully more people will start using it, improving its features. There's still some issues that I may raise in the repo, but they are beyond the scope of this review. Thanks @kthyng for your efforts too.
> 
> [![000200](https://user-images.githubusercontent.com/25530332/111016072-c43a6000-83a3-11eb-9da1-794a0468fd09.png) Click to see the video on twitter](https://twitter.com/milankloewer/status/1370561010856251393?s=20)

Many thanks @milankl , not only for this great application example but also for the time you spent on this review and the feedback you provided, which has led to major improvements of the package I feel. Please do continue to raise issues in the repo whenever needed. Thanks again

**Comment by gaelforget** [comment]
Hi @kthyng 

Am not sure what the next steps are but hoping we are close to completing the process. Please let me know if there is something else I should do at this stage. Thank you in advance

**Comment by kthyng** [comment]
@gaelforget Ack, sorry. I have been experimenting with labeling my email and I was too effective in hiding away my JOSS emails. Oops!

I see that both reviewers are satisfied with your submission, so I will go through some final steps now.

**Comment by kthyng** [comment]
@whedon generate pdf from branch josspaper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch josspaper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
@gaelforget What version do you want associated with this submission? Should it be updated?

Also, please archive your software at a place like Zenodo and check back here with the doi they give you. Please modify the metadata of the archive so that the title and author list exactly match your JOSS submission.

**Comment by kthyng** [comment]
@gaelforget A few things in your paper to fix. Let me know if you have questions:

1. Please check all of your inline references. If they are not meant to be parenthetical, leave off the [] around the reference so they are not parenthetical. This comes up a lot in the paper.
2. Check how you refer to the languages and other items through the paper. "Fortran" for example is all lower case one place but I believe should be "Fortran". 
3. Check your references in detail. One has "{ECCO}" in it and presumably shouldn't, your name sometimes has an umlaut but not always, and some words are not capitalized that should be like Lagrangian and Atlantic. You can preserve capitalization by adding {} around the words in the .bib file that should be kept as is. Please check for other issues, too.

**Comment by kthyng** [comment]
@whedon check references

**Comment by kthyng** [comment]
@whedon check references from branch josspaper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch josspaper
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1038/s41561-019-0333-7 is OK
- 10.5281/zenodo.1211363 is OK
- 10.1175/2009JPO4043.1 is OK
- 10.7910/DVN/RNXA2A is OK
- 10.7910/DVN/NXYKDW is OK
- 10.5194/gmd-8-3071-2015 is OK
- 10.5281/zenodo.3967889 is OK
- 10.5281/zenodo.2653669 is OK
- 10.5281/zenodo.4193582 is OK
- 10.1029/2020GL089137 is OK
- 10.5194/gmd-12-4955-2019 is OK
- 10.1029/2019GL085837 is OK
- 10.5281/zenodo.3995076 is OK
- 10.5281/zenodo.4517441 is OK
- 10.5194/gmd-2020-385 is OK

MISSING DOIs

- None

INVALID DOIs

- https://doi.org/10.1016/j.ocemod.2017.11.008 is INVALID because of 'https://doi.org/' prefix
```

**Comment by kthyng** [comment]
@gaelforget also fix that invalid doi please

**Comment by danielskatz** [comment]
👋 @gaelforget - This is waiting for you to make changes in your paper so that it can be published.

**Comment by gaelforget** [comment]
@whedon generate pdf from branch josspaper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch josspaper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: :point_left:

**Comment by gaelforget** [comment]
@whedon check references from branch josspaper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch josspaper
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1038/s41561-019-0333-7 is OK
- 10.5281/zenodo.1211363 is OK
- 10.1175/2009JPO4043.1 is OK
- 10.7910/DVN/RNXA2A is OK
- 10.7910/DVN/NXYKDW is OK
- 10.5194/gmd-8-3071-2015 is OK
- 10.5281/zenodo.3967889 is OK
- 10.5281/zenodo.2653669 is OK
- 10.5281/zenodo.4193582 is OK
- 10.1029/2020GL089137 is OK
- 10.5194/gmd-12-4955-2019 is OK
- 10.1016/j.ocemod.2017.11.008 is OK
- 10.1029/2019GL085837 is OK
- 10.5281/zenodo.3995076 is OK
- 10.5281/zenodo.4517441 is OK
- 10.5194/gmd-2020-385 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by gaelforget** [comment]
> @gaelforget also fix that invalid doi please

- Should be all set now on the references, I think -- unless there is a way, which I did not find, to remove parentheses around the year in e.g. `(Gaël Forget (2010), G. Forget et al. (2015), Gaël Forget (2018))` 
- `Fortran` is now capitalized throughout
- I changed a few cosmetic details (mostly moving the plots to page 2) following a final pass on the paper

**Comment by gaelforget** [comment]
> @gaelforget What version do you want associated with this submission? Should it be updated?
> 
> Also, please archive your software at a place like Zenodo and check back here with the doi they give you. Please modify the metadata of the archive so that the title and author list exactly match your JOSS submission.

Thanks for the very helpful directions

I would like version `v0.3.0`, which I released just before this post, to be associated with this submission. 

The corresponding reference, with edited metadata, is

Gael Forget. (2021, April 10). IndividualDisplacements.jl : a Julia package to simulate and study particle displacements within the climate system (Version v0.3.0). Zenodo. http://doi.org/10.5281/zenodo.4678742

**Comment by gaelforget** [comment]
> 👋 @gaelforget - This is waiting for you to make changes in your paper so that it can be published.

Hi @danielskatz 

Sorry for the lag on my end. Please find above my recent updates / replies. Look forward to completing the process if we are all set.

Thank you in advance

**Comment by gaelforget** [comment]
> @gaelforget Ack, sorry. I have been experimenting with labeling my email and I was too effective in hiding away my JOSS emails. Oops!
> 
> I see that both reviewers are satisfied with your submission, so I will go through some final steps now.

Hi @kthyng 

Thank you very much for the guidelines and all that you have done on this submission. Please find above responses to the issues listed. Thanks again

**Comment by kyleniemeyer** [comment]
@whedon generate pdf from branch josspaper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch josspaper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: :point_left:

**Comment by kyleniemeyer** [comment]
@whedon set 10.5281/zenodo.4678742 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4678742" target="_blank">10.5281/zenodo.4678742</a> is the archive.

**Comment by kyleniemeyer** [comment]
Hi @gaelforget, this is nearly ready for acceptance, but the parenthetical issues with the references throughout the paper remain.

When citing things, you do not need to include "e.g." or "see" or "for example"—this is implied whenever you cite one or more publications. But, this is more a preference, I suppose. The only real places that need to be fixed are when you have the nested parentheses like in the fourth paragraph; the appropriate syntax for grouped citations is `[@author1:2001; @author2:2001]`, which formats like -> "(Author1 et al., 2001; Author2 et al., 2002)".

I've made these changes directly in https://github.com/JuliaClimate/IndividualDisplacements.jl/pull/71, can you merge this?

**Comment by gaelforget** [comment]
> Hi @gaelforget, this is nearly ready for acceptance, but the parenthetical issues with the references throughout the paper remain.
> 
> When citing things, you do not need to include "e.g." or "see" or "for example"—this is implied whenever you cite one or more publications. But, this is more a preference, I suppose. The only real places that need to be fixed are when you have the nested parentheses like in the fourth paragraph; the appropriate syntax for grouped citations is `[@author1:2001; @author2:2001]`, which formats like -> "(Author1 et al., 2001; Author2 et al., 2002)".
> 
> I've made these changes directly in [JuliaClimate/IndividualDisplacements.jl#71](https://github.com/JuliaClimate/IndividualDisplacements.jl/pull/71), can you merge this?

Merged. Thanks @kyleniemeyer

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: :point_left:

**Comment by kyleniemeyer** [comment]
Hi @gaelforget, looks like there were still some issues, or perhaps I made mistakes: https://github.com/JuliaClimate/IndividualDisplacements.jl/pull/72

**Comment by gaelforget** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: :point_left:

**Comment by gaelforget** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02813/joss.02813/10.21105.joss.02813.pdf) :page_facing_up: :point_left:

**Comment by gaelforget** [comment]
> Hi @gaelforget, looks like there were still some issues, or perhaps I made mistakes: [JuliaClimate/IndividualDisplacements.jl#72](https://github.com/JuliaClimate/IndividualDisplacements.jl/pull/72)

Just did a couple more tweaks. Although I feel unsure what the desired rendering really is in some cases, this looks good to me now.

**Comment by kyleniemeyer** [comment]
@whedon accept

@gaelforget ok, looks good to me.

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1038/s41561-019-0333-7 is OK
- 10.5281/zenodo.1211363 is OK
- 10.1175/2009JPO4043.1 is OK
- 10.7910/DVN/RNXA2A is OK
- 10.7910/DVN/NXYKDW is OK
- 10.5194/gmd-8-3071-2015 is OK
- 10.5281/zenodo.3967889 is OK
- 10.5281/zenodo.2653669 is OK
- 10.5281/zenodo.4193582 is OK
- 10.1029/2020GL089137 is OK
- 10.5194/gmd-12-4955-2019 is OK
- 10.1016/j.ocemod.2017.11.008 is OK
- 10.1029/2019GL085837 is OK
- 10.5281/zenodo.3995076 is OK
- 10.5281/zenodo.4517441 is OK
- 10.5194/gmd-2020-385 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2263

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2263, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1386739657384370181) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2264
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02813](https://doi.org/10.21105/joss.02813)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
Congrats @gaelforget on your article's publication in JOSS!

Many thanks to @milankl and @visr for reviewing this, and @kthyng for editing.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02813/status.svg)](https://doi.org/10.21105/joss.02813)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02813">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02813/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02813/status.svg
   :target: https://doi.org/10.21105/joss.02813
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02813/status.svg)](https://doi.org/10.21105/joss.02813)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by gaelforget** [comment]
> Congrats @gaelforget on your article's publication in JOSS!
> 
> Many thanks to @milankl and @visr for reviewing this, and @kthyng for editing.

Awesome news! Thank you @kyleniemeyer 

Thank you @milankl and @visr -- your reviews definitely helped improve IndividualDisplacements.jl

Thank you also @kthyng and @kyleniemeyer for editing
